test/cast/number_test.exs

```diff
@@ -0,0 +1,66 @@
+defmodule OpenApiSpex.CastNumberTest do
+  use ExUnit.Case
+  alias OpenApiSpex.{Cast, Schema}
+  alias OpenApiSpex.Cast.{Error, Number}
+
+  defp cast(ctx), do: Number.cast(struct(Cast, ctx))
+
+  describe "cast/1" do
+    test "basics" do
+      schema = %Schema{type: :number}
+      assert cast(value: 1, schema: schema) == {:ok, 1}
```

Comment:
Since we're dealing with numbers, would it make sense to use the `===` operator?

Reply:
Good point, will change.

Reply:
Ok, an interesting effect of this that if a property of `type: :number` is fed a JSON integer, then we return the same integer as the cast value. Otherwise, it's a float.

It's just a slight inconsistency but it might cause confusion later on (esp. when paired with `is_float()` guards) if not documented.

We can 'coerce' into a float if we want to. WDYT?

Reply:
I suppose we could follow the Java rule that if there is a decimal point, then it is a float, otherwise it is an integer. The Jason and Poison JSON parsers both apply this rule, even though JSON (and JavaScript) don't have separate float and integer types.

```elixir
> is_integer(Jason.decode!("42"))
true
> is_integer(Poison.decode!("42"))
true
```

```javascript
> JSON.stringify(42)
"42"
> JSON.stringify(42.0)
"42"
```

It seems dangerous to me. What if the number was intended to be a float, but it just happens to be a whole number float? However, since Poison and Jason do it, I guess we can too.